presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CombineApproxPercentileFunctions.java

```diff
@@ -0,0 +1,296 @@
+                continue;
+            }
+            hasChanged = true;
+            List<RowExpression> percentileArray = new ArrayList<>();
+            Map<VariableReferenceExpression, Integer> variableToArrayIndex = new HashMap<>();
+            candidateList.forEach(candidate -> {
+                percentileArray.add(aggregationPercentileMap.get(candidate));
+                variableToArrayIndex.put(aggregationVariableMap.get(candidate), percentileArray.size());
+            });
+
+            RowExpression arrayExpression = call(
+                    functionAndTypeManager,
+                    "array_constructor",
+                    new ArrayType(percentileArray.get(0).getType()),
+                    percentileArray);
+            VariableReferenceExpression arrayVariableReference = context.getVariableAllocator().newVariable(arrayExpression);
+            // Add array to project node
+            sourceProjectAssignments.put(arrayVariableReference, arrayExpression);
+
+            // Create new aggregation
+            AggregationNode.Aggregation aggregationBeforeMerge = candidateList.get(0);
+            int percentilePosition = getPercentilePosition(aggregationBeforeMerge.getFunctionHandle());
+            List<RowExpression> newAggCallArguments = changePercentileArgument(aggregationBeforeMerge.getCall().getArguments(), arrayVariableReference, percentilePosition);
+            Type colType = aggregationBeforeMerge.getCall().getArguments().get(0).getType();
+            CallExpression approxPercentileCall = call(
+                    functionAndTypeManager,
+                    APPROX_PERCENTILE,
+                    new ArrayType(colType),
+                    newAggCallArguments);
+
+            AggregationNode.Aggregation newAggregation =
+                    new AggregationNode.Aggregation(
+                            approxPercentileCall,
+                            aggregationBeforeMerge.getFilter(),
+                            aggregationBeforeMerge.getOrderBy(),
+                            aggregationBeforeMerge.isDistinct(),
+                            aggregationBeforeMerge.getMask());
+            // Get variable reference to the new aggregation
+            VariableReferenceExpression newVariableReference = context.getVariableAllocator().newVariable(newAggregation.getCall());
+            aggregations.put(newVariableReference, newAggregation);
+
+            variableToArrayIndex.forEach((variable, index) -> {
+                RowExpression elementAtExpression = call(
+                        functionAndTypeManager,
+                        "element_at",
+                        colType,
+                        ImmutableList.of(newVariableReference, constant((long) index, BIGINT)));
+                outputProjectAssignments.put(variable, elementAtExpression);
+                combinedAggVariableReference.add(variable);
+            });
+        }
+
+        if (!hasChanged) {
```

Comment:
Make it more readable like newAggregationsGenerated so the variable is more descriptive

Reply:
Move the check to earlier stage where after we check if there are candidate aggregations to merge, and without using an additional bool.